### DISCUSSION TITLE
How to get start with Wgpu-HAL?

### DISCUSSION THREAD

**Discussion by Local-Trash** [gfx-rs/wgpu#5147] [Q&A]
I'm trying to learn `wgpu-hal`, so I can understand how Low Level Graphics APIs. It is also cross platform. I tried going through the examples, but they are outdated. How would I fill a simple window with one color? Thank you.

**Comment by cwfitzgerald** [comment]
I would highly encourage against using wgpu-hal directly. It's basically an implementation detail of wgpu and there are zero learning resources for it and documentation is spotty. I would suggest using wgpu directly and looking at one of the tutorials mentioned in the readme.

As for the examples, make sure you're using the examples from the relevant branch. If you're targeting v0.19, use the examples from that branch.
